vulnerabilities/importers/ruby.py

```diff
@@ -7,132 +7,147 @@
+    """
+    if schema_type == "gems":
         package_name = record.get("gem")
-        if not package_name:
-            return
-
-        if "cve" in record:
-            cve_id = "CVE-{}".format(record["cve"])
-        else:
-            return
-
-        publish_time = parse(record["date"]).replace(tzinfo=UTC)
-        safe_version_ranges = record.get("patched_versions", [])
-        # this case happens when the advisory contain only 'patched_versions' field
-        # and it has value None(i.e it is empty :( ).
-        if not safe_version_ranges:
-            safe_version_ranges = []
-        safe_version_ranges += record.get("unaffected_versions", [])
-        safe_version_ranges = [i for i in safe_version_ranges if i]
-
-        if not getattr(self, "pkg_manager_api", None):
-            self.pkg_manager_api = RubyVersionAPI()
-        all_vers = self.pkg_manager_api.get(package_name, until=publish_time).valid_versions
-        safe_versions, affected_versions = self.categorize_versions(all_vers, safe_version_ranges)
-
-        impacted_purls = [
-            PackageURL(
-                name=package_name,
-                type="gem",
-                version=version,
-            )
-            for version in affected_versions
-        ]
-
-        resolved_purls = [
-            PackageURL(
-                name=package_name,
-                type="gem",
-                version=version,
-            )
-            for version in safe_versions
-        ]
+        purl = PackageURL(type="gem", name=package_name)
 
-        references = []
-        if record.get("url"):
-            references.append(Reference(url=record.get("url")))
+        return AdvisoryData(
+            aliases=get_aliases(record),
+            summary=get_summary(record),
+            affected_packages=get_affected_packages(record, purl),
+            references=get_references(record),
+            date_published=get_publish_time(record),
+        )
 
+    elif schema_type == "rubies":
+        engine = record.get("engine")  # engine enum: [jruby, rbx, ruby]
+        purl = PackageURL(type="ruby", name=engine)
         return AdvisoryData(
-            summary=record.get("description", ""),
-            affected_packages=nearest_patched_package(impacted_purls, resolved_purls),
-            references=references,
-            vulnerability_id=cve_id,
+            aliases=get_aliases(record),
+            summary=get_summary(record),
+            affected_packages=get_affected_packages(record, purl),
+            references=get_references(record),
+            date_published=get_publish_time(record),
         )
 
-    @staticmethod
-    def categorize_versions(all_versions, unaffected_version_ranges):
 
-        for id, elem in enumerate(unaffected_version_ranges):
-            unaffected_version_ranges[id] = VersionRange.from_scheme_version_spec_string(
-                "semver", elem
+def get_affected_packages(record, purl):
```

Comment:
@ziadhany please add docstring

Reply:
Also please add some unit tests for this function with some corner cases

Reply:
Done